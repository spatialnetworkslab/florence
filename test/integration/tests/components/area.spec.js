/// <reference types="Cypress" />

context('Assertions', () => {
  beforeEach(() => {
    cy.visit('/area')
  })

  describe('Area Tests', () => {
    it('[render] area rendered', () => {
      cy.get('.area').should('have.length', 1)
    })

    it('[render] area snapshot correctly', () => {
      cy.get('svg').snapshot({ name: 'svg with 1 area mark' })
    })

    it('[interaction] hovering over area changes color', () => {
      cy.get('.area').should('have.attr', 'fill', 'steelblue')
      cy.get('.area').trigger('mousemove')
      cy.get('.area').should('have.attr', 'fill', '#d5896f')
      cy.get('.left-col').trigger('mousemove')
      cy.get('.area').should('have.attr', 'fill', 'steelblue')
    })
  })
})
