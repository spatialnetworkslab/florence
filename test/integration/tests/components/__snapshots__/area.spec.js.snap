exports[`Assertions > Area Tests > [render] area snapshot correctly #0`] = `
<svg width="700" height="500">
  <defs>
    <clipPath id="clip-sc0">
      <rect x="0" y="0" width="700" height="500"></rect>
    </clipPath>
    <clipPath id="clip-sc0-data">
      <rect x="30" y="20" width="650" height="450" fill="white"></rect>
    </clipPath>
  </defs>
  <g class="section" clip-path="url(#clip-sc0)">
    <path
      class="area"
      d="M30,470L64.21,470L98.42,470L132.63,470L166.84,470L201.05,470L235.26,470L269.47,470L303.68,470L337.89,470L372.11,470L406.32,470L440.53,470L474.74,470L508.95,470L543.16,470L577.37,470L611.58,470L645.79,470L680,470L680,395.82L645.79,227.69L611.58,390.88L577.37,133.74L543.16,336.48L508.95,385.93L474.74,143.63L440.53,39.78L406.32,227.69L372.11,198.02L337.89,232.64L303.68,212.86L269.47,39.78L235.26,376.04L201.05,207.91L166.84,69.45L132.63,20L98.42,257.36L64.21,198.02L30,331.54Z"
      fill="steelblue"
      stroke="none"
      stroke-width="0"
      opacity="1"
    ></path>
    <g class="x-axis">
      <g class="line-layer">
        <path
          class="line"
          d="M64.21,470.45L64.21,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M132.63,470.45L132.63,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M201.05,470.45L201.05,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M269.47,470.45L269.47,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M337.89,470.45L337.89,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M406.32,470.45L406.32,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M474.74,470.45L474.74,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M543.16,470.45L543.16,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M611.58,470.45L611.58,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M680,470.45L680,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
      </g>
      <g class="label-layer">
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="64.21052631578947"
          opacity="1"
          transform="
          rotate(0, 
          64.21052631578947, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          2
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="132.6315789473684"
          opacity="1"
          transform="
          rotate(0, 
          132.6315789473684, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          4
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="201.05263157894737"
          opacity="1"
          transform="
          rotate(0, 
          201.05263157894737, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          6
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="269.4736842105263"
          opacity="1"
          transform="
          rotate(0, 
          269.4736842105263, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          8
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="337.89473684210526"
          opacity="1"
          transform="
          rotate(0, 
          337.89473684210526, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          10
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="406.3157894736842"
          opacity="1"
          transform="
          rotate(0, 
          406.3157894736842, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          12
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="474.7368421052632"
          opacity="1"
          transform="
          rotate(0, 
          474.7368421052632, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          14
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="543.1578947368421"
          opacity="1"
          transform="
          rotate(0, 
          543.1578947368421, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          16
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="611.5789473684209"
          opacity="1"
          transform="
          rotate(0, 
          611.5789473684209, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          18
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="680"
          opacity="1"
          transform="
          rotate(0, 
          680, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          20
        </text>
      </g>
    </g>
    <g class="y-axis">
      <g class="line-layer">
        <path
          class="line"
          d="M29.54,470L23.96,470"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,420.55L23.96,420.55"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,371.1L23.96,371.1"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,321.65L23.96,321.65"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,272.2L23.96,272.2"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,222.75L23.96,222.75"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,173.3L23.96,173.3"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,123.85L23.96,123.85"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,74.4L23.96,74.4"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,24.95L23.96,24.95"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
      </g>
      <g class="label-layer">
        <text
          class="label"
          y="470"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          470)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          0
        </text>
        <text
          class="label"
          y="420.54945054945057"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          420.54945054945057)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          10
        </text>
        <text
          class="label"
          y="371.0989010989011"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          371.0989010989011)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          20
        </text>
        <text
          class="label"
          y="321.64835164835165"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          321.64835164835165)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          30
        </text>
        <text
          class="label"
          y="272.1978021978022"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          272.1978021978022)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          40
        </text>
        <text
          class="label"
          y="222.7472527472527"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          222.7472527472527)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          50
        </text>
        <text
          class="label"
          y="173.29670329670327"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          173.29670329670327)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          60
        </text>
        <text
          class="label"
          y="123.84615384615383"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          123.84615384615383)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          70
        </text>
        <text
          class="label"
          y="74.39560439560441"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          74.39560439560441)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          80
        </text>
        <text
          class="label"
          y="24.945054945054927"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          24.945054945054927)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          90
        </text>
      </g>
    </g>
  </g>
</svg>
`;
